src/NHibernate/LockMode.cs

```diff
@@ -11,11 +12,13 @@ namespace NHibernate
 	/// wish to explicitly specify lock levels.
 	/// </remarks>
 	[Serializable]
-	public sealed class LockMode
+	public sealed class LockMode : IDeserializationCallback
```

Comment:
I have missed that class in my first commit.

Reply:
NHibernate Core Distributed Cache Library line 179 gets key with key.ToString() when i see this one I can say that we might have been used ToString() to compare filters. I think the order of these values are not important as you mentioned ? Am i wrong?

`private string GetCacheKey(object key)
		{
			var keyAsString = AppendHashcodeToKey
				? string.Concat(_cacheKeyPrefix, _fullRegion, ":", key.ToString(), "@", key.GetHashCode())
				: string.Concat(_cacheKeyPrefix, _fullRegion, ":", key.ToString());

			if (_maxKeySize < keyAsString.Length)
			{
				Log.Info(
					"Computing a hashed key for too long key '{0}'. This may cause collisions resulting into additional cache misses.",
					key);
				// Hash it for respecting max key size. Collisions will be avoided by storing the actual key along
				// the object and comparing it on retrieval.
				using (var hasher = new SHA256Managed())
				{
					var bytes = Encoding.UTF8.GetBytes(keyAsString);
					var computedHash = Convert.ToBase64String(hasher.ComputeHash(bytes));
					if (computedHash.Length <= _maxKeySize)
						return computedHash;

					if (!_hasWarnedOnHashLength)
					{
						// No lock for this field, some redundant logs will be less harm than locking.
						_hasWarnedOnHashLength = true;
						Log.Warn(
							"Hash computed for too long keys are themselves too long. They will be truncated, further " +
							"increasing the risk of collision resulting into additional cache misses. Consider using a " +
							"cache supporting longer keys. Hash length: {0}; max key size: {1}",
							computedHash.Length, _maxKeySize);
					}

					keyAsString = computedHash.Substring(0, _maxKeySize.Value);
				}
			}

			return _keySanitizer != null ? _keySanitizer(keyAsString) : keyAsString;
		}`